### ISSUE TITLE
Visual Classification via Description from Large Language Models

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#208]
# TL; DR

<img width="735" alt="image" src="https://user-images.githubusercontent.com/46308407/217687371-05eab410-ffac-4804-a88d-fe782f1326bc.png">

- Collect visual clues for better (accuracy, interpretability, fairness) visual recognition.

# Motivation

- Previous V-L frameworks typically use prompted category name as the only textual features, which limits their visual understanding as well as interpretability.

# Formulation

<img width="738" alt="image" src="https://user-images.githubusercontent.com/46308407/217687587-c8a973a2-27ca-41a2-82f6-56c1e30fad41.png">

- The key idea is to build a set of ***descriptors*** that is useful to identify certain object category.
- The descriptors are automatically generated by a powerful language model (*e.g., GPT-3*) by prompting as following:
<img width="622" alt="image" src="https://user-images.githubusercontent.com/46308407/217688208-ffd374b3-a9f7-4228-883a-a5af489ace70.png">
<img width="736" alt="image" src="https://user-images.githubusercontent.com/46308407/217688337-4a6b13d0-d3d5-4418-af70-a05cc1bfa09c.png">

- After descriptors has been generated, simply average their scores to make the final output (*ensemble*).

# Evaluation

### Accuracy

<img width="729" alt="image" src="https://user-images.githubusercontent.com/46308407/217689038-a7ad3639-42f0-41b3-9a9a-d061ac84bab7.png">

- Descriptor-based ensemble improves image recognition performance across architectures and benchmarks.

### Explainability

<img width="722" alt="image" src="https://user-images.githubusercontent.com/46308407/217689182-472dec67-5f52-457b-8e92-add5e5630ac0.png">

- Because we aggregate scores for each attributes, we can explicitly find out where the model went wrong, and why the model has made such decision.

### Fairness

<img width="673" alt="image" src="https://user-images.githubusercontent.com/46308407/217689584-a4d6ad37-9a56-4997-ab41-4bd4039f875c.png">

- We can detect and fix undesired biases in V-L frameworks.
- For each concept we are dealing with, we can generate a more diverse set of descriptors that hopefully embrace a wider spectrum of culture and heritage, which can possibly remedy some innate problems of foundation models.

### External Knowledge Injection

<img width="723" alt="image" src="https://user-images.githubusercontent.com/46308407/217689300-21f5d0ed-1111-4c07-b77f-0dbfb6a4370e.png">

- V-L models must directly associate visual information to its corresponding language concept.
- If the language model is updated to the expansion of our language practice, we can empower the V-L framework with the additional clues.
- To put it simply, CLIP becomes capable of recognizing visual concepts that has not existed during its training, thanks to the rich descriptors of provided by the language model.

# Inspiration

- For some time, I have thought that using a language model (or possibly a captioner) would be a more scalable and elegant choice for explainable AI. This work (though indirectly) tackles such problem in a very successful way.
- The method is simple yet effective, bringing so many benefits in terms of accuracy, interpretability and fairness.
- That being said, there are some limitations as well. Most importantly, this framework requires a fixed set of class categories, rendering it unsuitable for open set recognition and more generalized visual reasoning.
- Still, obtaining performance and interpretability with such a simple remedy is very attractive. Exciting room for further improvements.
